boilerplate/app/components/screen/screen.tsx

```diff
@@ -32,6 +32,37 @@ function ScreenWithScrolling(props: ScreenProps) {
   const backgroundStyle = props.backgroundColor ? { backgroundColor: props.backgroundColor } : {}
   const insetStyle = { paddingTop: props.unsafe ? 0 : insets.top }
 
+  // The followings for <Screen preset='auto'/>
+  // This will automatically disables scrolling if content fits the screen.
+  const { height } = Dimensions.get('window')
+  const screenHeight = React.useRef(null)
+  const [scrollEnabled, setScrollEnabled] = React.useState(true)
+
+  const updateScrollState = () => {
+    if (props.preset === 'auto'){
+      // check whether if content fits the screen
+      const contentFitsScreen = screenHeight.current > height * presets.auto.offset.percent - presets.auto.offset.point
```

Comment:
I could be misreading this, but doesn't this variable indicate that the content _doesn't_ fit the screen anymore, and thus we need to enable scroll? Shouldn't it be something like `contentOverflowsScreen`?

Reply:
Nope, it indicates exactly the opposite. More precisely: Content is smaller than screen size.

How about using **ContentFitsWithinScreen**? Does it describe better in English?

Reply:
I guess what I'm confused by is the `height` variable comes from `Dimensions.get("window")`, so the logic seems to be:

contentFitsScreen = the screen's height is _larger_ than the window's height

In my mind, it should be the opposite, right?

`height` indicates the window height (see line 37).

In fact, I don't see anything about the content height at all.

Can you walk me through the logic here? Sorry this is confusing to me!

Reply:
All right, I've reviewed the code again. You are right. The variable name is wrong. Sorry about that!

Even `screenHeight` name is kinda wrong. You even mentioned that at one point in your stream. It's confusing because it can either mean the height of the device screen and the height of the component too.

So we better go with `scrollViewHeight` name there which is more clear and even looks better that all scroll related variables are grouped together. What do you think?

For `contentFitsScreen` the declaring really mean the opposite, I just did not catch this until I reviewed again. I'm gonna change the angle bracket there, and fix those two ifs according to it.